testsuite/testcases/src/compatibility_test.rs

```diff
@@ -33,105 +23,28 @@ impl Test for SimpleValidatorUpgrade {
-    Ok(statsum)
-}
-
-fn upgrade_and_gather_stats(
-    ctxa: NetworkContextSynchronizer,
+async fn upgrade(
+    ctxa: NetworkContextSynchronizer<'_>,
     // upgrade args
     validators_to_update: &[PeerId],
     version: &Version,
     wait_until_healthy: bool,
     delay: Duration,
     max_wait: Duration,
-    // traffic args
-    nodes: &[PeerId],
-) -> Result<Option<TxnStats>> {
-    let upgrade_done = Arc::new(AtomicBool::new(false));
-    let emitter_ctx = ctxa.clone();
-    let mut stats_result: Result<Option<TxnStats>> = Ok(None);
-    let mut upgrade_result: Result<()> = Ok(());
-    tokio_scoped::scope(|scopev| {
-        // emit trafic and gather stats
-        scopev.spawn(async {
-            info!("upgrade_and_gather_stats traffic thread start");
-            let (emitter, emit_job_request, source_account) = {
-                let mut ctx_locker = emitter_ctx.ctx.lock().await;
-                let ctx = ctx_locker.deref_mut();
-                let emit_job_request = ctx.emit_job.clone();
-                let rng = SeedableRng::from_rng(ctx.core().rng()).unwrap();
-                let (emitter, emit_job_request) = match create_emitter_and_request(
-                    ctx.swarm.clone(),
-                    emit_job_request,
-                    nodes,
-                    rng,
-                )
-                .await
-                {
-                    Ok(parts) => parts,
-                    Err(err) => {
-                        stats_result = Err(err);
-                        return;
-                    },
-                };
-                let source_account = ctx.swarm.read().await.chain_info().root_account;
-                (emitter, emit_job_request, source_account)
-                // release lock on network context
-            };
-            let upgrade_traffic_chunk_duration = Duration::from_secs(15);
-            info!("upgrade_and_gather_stats traffic thread 1");
-            stats_result = stat_gather_task(
-                emitter,
-                emit_job_request,
-                source_account,
-                upgrade_traffic_chunk_duration,
-                upgrade_done.clone(),
-            )
-            .await;
-            info!("upgrade_and_gather_stats traffic thread done");
-        });
-        // do upgrade
-        scopev.spawn(async {
-            info!("upgrade_and_gather_stats upgrade thread start");
-            upgrade_result = batch_update_gradually(
-                ctxa,
-                validators_to_update,
-                version,
-                wait_until_healthy,
-                delay,
-                max_wait,
-            )
-            .await;
-            info!("upgrade_and_gather_stats upgrade thread 1");
-            upgrade_done.store(true, Ordering::Relaxed);
-            info!("upgrade_and_gather_stats upgrade thread done");
-        });
-    });
-
+) -> Result<()> {
+    info!("upgrade_and_gather_stats upgrade thread start");
```

Comment:
nit here but this is not really happening in a separate thread anymore is it?

We can just remove this function entirely since its just calling another batch update gradually